### PR TITLE
bump gopkg.in/yaml.v2

### PR DESCRIPTION
**Pull request by SalDaniele**
Upgrade gopkg.in/yaml.v2 prior to v2.2.4. Currently this outdated version is indirectly references as follows:

github.com/stretchr/testify requires gopkg.in/yaml.v2 v2.2.2

github.com/go-task/slim-sprig v0.0.0-20210107165309-348f09dbbbc0 requires gitub.com/stretchr/testify v1.5.1, and is required by a number of packages.

To upgrade from yaml <v2.2.4, ginkgo must be migrated to v2.

Additionally, bump the versions of the following to remove indirect dependencies on outdated version of ginkgo, gomega, and slim-sprig:

github.com/onsi/gomega
github.com/containernetworking/cni
github.com/containernetworking/plugins

### CODE REVIEW

**Comment by SalDaniele** [comment]
Is it possible to get this reviewed / merged? This update is required to fix https://issues.redhat.com/browse/OCPBUGS-6473

**Comment by SalDaniele** [comment]
cc @adrianchiris @SchSeba

**Comment by SalDaniele** [comment]
@wizhao

**Comment by SchSeba** [comment]
there is an issue with the go-lint but there is no output :(

I will need to try and run it locally with the same version to see if it also failed on my laptop

File: go.mod [hunk 1]
```diff
@@ -3,26 +3,28 @@ module github.com/k8snetworkplumbingwg/sriov-cni
...
 
 require (
-	github.com/containernetworking/cni v1.1.2
-	github.com/containernetworking/plugins v1.1.1
-	github.com/onsi/ginkgo v1.16.4
-	github.com/onsi/gomega v1.17.0
-	github.com/stretchr/testify v1.5.1
-	github.com/vishvananda/netlink v1.1.1-0.20210330154013-f5de75959ad5
-	golang.org/x/net v0.7.0
+	github.com/containernetworking/cni v1.1.3-0.20230206011450-121798a08dcc
```
Comment (zeeke) [inline go.mod:6]: nit: Is bumping `github.com/containernetworking/cni` strictly necessary?
I'd prefer to not setting this dep to a non released one, if possible.

**Comment by zeeke** [comment]
@e0ne, @adrianchiris can we have some clues about Mellanox CI failure?

File: go.mod (hunk 1 above)
Comment (SalDaniele) [reply go.mod:6]: Unfortunately yes, as the current release of  github.com/containernetworking/cni (1.1.2) pulls in an older version of ginkgo, which by extension pulls in the outdated package.

i.e. cni@v1.1.2 -> ginkgo@v2.1.3->slim-sprig->testify@v1.5.1->yaml.v2@v2.2.2

**Comment by SalDaniele** [comment]
@adrianchiris @e0ne Looks like the Mellanox CI is no longer auto-failing, but it has been stuck on pending for over 24 hours. Any idea as to how we can get this PR merged?

File: go.mod [hunk 2]
```diff
@@ -3,26 +3,28 @@ module github.com/k8snetworkplumbingwg/sriov-cni
...
+	github.com/containernetworking/plugins v1.2.1-0.20230314104937-3bc00017e302
```
Comment (adrianchiris) [inline go.mod:7]: not a fan of bumping those two (cni and plugins) to non released version.

if those drag in the problematic dependency, we can overwrite in a replace directive [1] WDYT? 

[1] https://go.dev/ref/mod#go-mod-file-replace

Comment (SalDaniele) [reply go.mod]: Hi @adrianchiris, thanks for the review. Makes sense to me, just pushed these changes.

**Review by adrianchiris — APPROVED** [review]

**Comment by SalDaniele** [comment]
Hi @zeeke is it possible to get this merged? Looks like the Mellanox CI is running / passing

**Pull request merged**